### PR TITLE
Updated to gitlab v4 api

### PR DESCRIPTION
**Pull request by skirill**
- Based on bugzilla login support (previous pull request)
- Removes fields not existing in v4 API
- Adds 'created_at' support for issues and comments to preserve time info
- assignee_ids is now a list
- iid should be used to refer to issue ID

Checked against Gitlab 10.1.1

### CODE REVIEW

**Comment by xmunoz** [comment]
I'll review this and other PR once the first one is merged.

File: bugzilla2gitlab/models.py [hunk 1]
```diff
@@ -44,17 +44,16 @@ def save(self):
             comment.save()
 
         # close the issue in GitLab, if it is resolved in Bugzilla
-        if self.issue.status == "RESOLVED":
+        if self.issue.status in ["RESOLVED", "VERIFIED", "CLOSED"]:
```
Comment (xmunoz) [inline bugzilla2gitlab/models.py]: Since there exists the possibility for many bugzilla statuses that map to "Closed" in gitlab, I'd like to see this as a configuration variable in `defaults.yml`

File: bugzilla2gitlab/models.py [hunk 2]
```diff
@@ -44,17 +44,16 @@ def save(self):
...
 
 
 class Issue(object):
     '''
     The issue model
     '''
-    required_fields = ["sudo", "title", "description", "status"]
-    data_fields = ["sudo", "title", "description", "status", "assignee_id", "milestone",
-                   "labels"]
+    required_fields = ["title", "description"]
```
Comment (xmunoz) [inline bugzilla2gitlab/models.py]: Here and other places you remove references to `sudo`. It doesn't look like `sudo` has been deprecated in the v4 API. Why did you remove it?

File: bugzilla2gitlab/utils.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def markdown_table_row(key, value):
     '''
     Create a row in a markdown table.
     '''
-    return "| {} | {} |\n".format(key, value)
+    return "| " + key + " | " + value + " |\n"
```
Comment (xmunoz) [inline bugzilla2gitlab/utils.py]: Why was this changed? Please revert to use the "new" format style.

File: bugzilla2gitlab/models.py (hunk 1 above)
Comment (skirill) [reply bugzilla2gitlab/models.py]: Done

File: bugzilla2gitlab/utils.py (hunk 3 above)
Comment (skirill) [reply bugzilla2gitlab/utils.py]: When table row contains non-ascii letters this causes an exception (in my case I had an attachment with non-ascii description but it might be also localized date string etc.). Now I see that it would be enough to explicitly make formatting template a unicode string: 
`    return u"| {} | {} |\n".format(key, value)`
Pushed this variant, please check.

**Comment by skirill** [comment]
Checks are failing because this branch bases on older version of bugzilla login branch where flake8 cleanup was done. I can create another "clean" branch if you wish (once review is over). Or you can merge manually and take includes from my version of util.py and models.py

**Comment by xmunoz** [comment]
I just did it on the command line, thanks!
